src/providers/details/tabs/Hosts/components/VSphereNetworkModal.tsx

```diff
@@ -0,0 +1,103 @@
+import { type FC, type ReactNode, useCallback, useState } from 'react';
+import { AlertMessageForModals } from 'src/modules/Providers/modals/components/AlertMessageForModals';
+import { useForkliftTranslation } from 'src/utils/i18n';
+
+import ModalForm from '@components/ModalForm/ModalForm';
+import type { NetworkAdapters, V1beta1Provider } from '@kubev2v/types';
+import { Form, ModalVariant } from '@patternfly/react-core';
+
+import { getSelectedInventoryHostNetworkTriples } from './utils/getSelectedInventoryHostNetworkTriples';
+import { onSaveHost } from './utils/onSaveHost';
+import type { InventoryHostNetworkTriple } from './utils/types';
+import { validatePassword, validateUsername } from './utils/validators';
+import HostsNetworksSelect from './HostsNetworksSelect';
+import HostsNetworksSetPassword from './HostsNetworksSetPassword';
+import HostsNetworksSetUserName from './HostsNetworksSetUserName';
+
+import './VSphereNetworkModal.style.css';
+
+type VSphereNetworkModalProps = {
+  provider: V1beta1Provider;
+  data: InventoryHostNetworkTriple[];
+  selectedIds: string[];
+};
+
+const VSphereNetworkModal: FC<VSphereNetworkModalProps> = ({ data, provider, selectedIds }) => {
+  const { t } = useForkliftTranslation();
+  const [alertMessage, setAlertMessage] = useState<ReactNode>(null);
+  const [network, setNetwork] = useState<NetworkAdapters>(undefined as unknown as NetworkAdapters);
+  const [username, setUsername] = useState<string>(undefined as unknown as string);
+  const [password, setPassword] = useState<string>(undefined as unknown as string);
```

Comment:
Why all of those assertions and double assertions? Undefined is the same as not putting anything. You can also use null if you want, without assertions, most of the time.